code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/process/ServiceActivate.java

```diff
@@ -211,10 +213,16 @@ public Object call() throws Exception {
         return ntwkIds;
     }
 
-    protected Long getImage(String imageUuid) {
+    protected Long getImage(String imageUuid, Object registryCredentialId) {
```

Comment:
Its supposed to be Long, but I put object to simply avoid casting (and not null checks) when fetching this field from launchConfig. As objectManager accepts type Object in setFields.